dev-haskell/fay/fay-0.24.2.0.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# ebuild generated by hackport 0.6.7.9999
+
+CABAL_FEATURES="lib profile haddock hoogle hscolour"
+inherit haskell-cabal
+
+DESCRIPTION="A compiler for Fay, a Haskell subset that compiles to JavaScript"
+HOMEPAGE="https://github.com/faylang/fay/wiki"
+SRC_URI="https://hackage.haskell.org/package/${P}/${P}.tar.gz"
+
+LICENSE="BSD"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+# See [issue upstream][1] for resolving circular dependency
+# [1]: https://github.com/faylang/fay/issues/473
+RESTRICT="test"
```

Comment:
Could you please explain to me the addition of `IUSE="test"` and `RESTRICT="test"` with the custom `src_test()` phase below?

From what I see this ebuild is trying to disable the test phase, in which case I think the test USE flag should be removed. If that is not the case, I think you need to change this to `RESTRICT="!test? ( test )"` to correctly interoperate with the `IUSE="test"` being enabled or disabled.

Feel free to seek comment from others on this!

Reply:
So the thinking here: I spent some time enabling the test, and made *some* progress (I think I actually succesfully ran the test), but I'm unable to push the change due to the circular dependency issue.

Rather than removing all the code in the ebuild needing to enable/run the test, I opted to simply `RESTRICT` it and cite the upstream ticket. Hopefully in the future the upstream ticket can be closed and we can re-enable the test.

If there is a preference to simply remove all the test stuff entirely, let me know and I can do this.

Reply:
The comment also does not explain what kind o circular dependency is here. If it's a build-time circular dependenci `RESTRICT=test` will not be enough. `USE=test` will still cause the same circular dependency problem.

Reply:
Hrm, i think the circular dependency was fixed when they bumped the version (I'm response to my issue, whoot!) But I think the tests still don't work.

I'll dig into this and fix the comment

Reply:
Ok, the circular dependency has been fixed upstream, but it actually causes most tests to fail. I've updated the comment to make this clear, and also reference the upstream ticket.